docs/articles/documentation/recipes/best-practices/best-practices.md

```diff
@@ -0,0 +1,332 @@
+        this.submitButton            = Selector('button').withAttribute('data-testid', 'submit-button');
+    }
+
+    async selectFeature(number) {
+        await t.click(this.importantFeaturesLabels.nth(number));
+    }
+
+    async clickSubmit() {
+        await t.click(this.submitButton);
+    }
+
+    async typeName(name) {
+        await t.typeText(this.nameInput, name);
+    }
+}
+
+export default new Page();
+```
+
+The page object holds references to the desired elements on the page. Common operations are defined as an object's method.
+
+This useful abstraction improves the flexibility of your tests - if the UI changes, change one file to update all the references. A test with this model can look like this:
+
+```js
+import page from './page-model'
+
+fixture `Use a Page Model`
+    .page `https://devexpress.github.io/testcafe/example`;
+
+
+test('Use a Page Model', async () => {
+
+    await page.selectFeature(2);
+    await page.typeName('Peter');
+    await page.clickSubmit();
+
+});
+```
+
+The page model object handles the identification of items and common operations. Tests are then more readable and less brittle because there is no duplicate code. Notice how this test doesn't require a test controller (`t`), since the page object handles all the actions.
+
+The [Page Model](../../guides/concepts/page-model.md) article covers the use of page objects in greater detail.
+
+A code example is available in the [testcafe-examples](https://github.com/DevExpress/testcafe-examples/tree/master/examples/use-page-model) repository.
+
+## Use of Roles for Login
+
+Handle authentication during your tests with [User Roles](../../guides/advanced-guides/authentication.md#user-roles). Roles allow you to wrap authentication logic and credentials in a reusable object.
+
+Define roles with the [Role](../../reference/test-api/role/constructor.md) constructor and include them in your tests with the [t.useRole](../../reference/test-api/testcontroller/userole.md) method. The following example makes use of two user roles:
+
+```js
+import { Role } from 'testcafe';
+
+const regularUser = Role('http://example.com/login', async t => {
+    await t
+        .typeText('#login', 'TestUser')
+        .typeText('#password', 'testpass')
+        .click('#sign-in');
+});
+
+const admin = Role('http://example.com/login', async t => {
+    await t
+        .typeText('#login', 'Admin')
+        .typeText('#password', 'adminpass')
+        .click('#sign-in');
+
+fixture `My Fixture`;
+
+test('My test', async t => {
+    await t
+        .useRole(role);
+});
+```
+
+See the [Authentication](../../guides/advanced-guides/authentication.md) article for more in-depth information.
+
+## File Structure
+
+Follow these guidelines to keep your test structure manageable and "clean":
+
+* Use a page model to store Selectors and compound actions that are used often across your app. For instance, put all actions that are necessary to perform an action into one reusable function.
```

Comment:
> that are used often across your app.

Is the following word order also correct: *that are used across your app often*?

Because I first read your version as `that are used, often across your app.` Like often across your app, sometimes not.

Reply:
> Use a page model to store Selectors and compound actions that are used across your app often. For instance, a page model function can contain all steps that are necessary to perform an action.